src/core/harpoon.ml

```diff
@@ -304,14 +304,130 @@ module Tactic (T : TacticContext) = struct
       ; solution = None
       }
     in
-    T.remove_current_subgoal ();
-    T.add_subgoal new_state;
+    tctx.remove_current_subgoal ();
+    tctx.add_subgoal new_state;
     prepend_commands
       [ Comp.IH (m, name) ]
       (Comp.incomplete_proof new_state)
     |> solve' g
 end
 
+module Automation = struct
+  open Comp
+
+  type t = unit proof_state -> Tactic.tactic_context -> bool
```

Comment:
This is *almost* the same as `Tactic.t`, so perhaps it's worthwhile to generalize `Tactic.t` to take a type parameter. Then "traditional" tactics have type `unit t` and automatic tactics have type `bool t`.